docs/src/test-merge-report-js.md

```diff
@@ -39,3 +39,178 @@ npx playwright merge-reports path/to/all-blob-reports --reporter html
+
+The logic of downloading blob report from GitHub Actions Artifacts has to be updated too as the merge workflow can not access artifacts from the test workflow directly. You can use the following custom action for that:
+
+```yaml
+name: 'Download blob report'
+description: 'Download blob report from GitHub artifacts'
+inputs:
+  name:
+    description: 'Name of the artifact to download'
+    required: true
+    type: string
+    default: 'blob-report'
+  path:
+    description: 'Directory with downloaded artifacts'
+    required: true
+    type: string
+    default: 'blob-report'
+runs:
+  using: "composite"
+  steps:
+    - name: Download blob report
+      uses: actions/github-script@v6
+      with:
+        script: |
+          const { data } = await github.rest.actions.listWorkflowRunArtifacts({
+            ...context.repo,
+            run_id: context.payload.workflow_run.id
+          });
+          const name = '${{ inputs.name }}';
+          const report = data.artifacts.filter(a => a.name === name)[0];
+          const result = await github.rest.actions.downloadArtifact({
+            ...context.repo,
+            artifact_id: report.id,
+            archive_format: 'zip'
+          });
+          const fs = require('fs');
+          fs.writeFileSync(`${name}.zip`, Buffer.from(result.data));
+    - name: Unzip blob report
+      shell: bash
+      run: unzip ${{ inputs.name }}.zip -d ${{ inputs.path }}
+```
+
+Putting it all together:
+
+```yaml
+name: Publish Test Results
+on:
+  workflow_run:
+    workflows: ["Playwright Tests"]
+    types:
+      - completed
+jobs:
+  merge-reports:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 16
+
+    - run: npm ci
+      env:
+        DEBUG: pw:install
+        PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1
+
+    - name: Download blob report artifact
+      uses: ./.github/actions/download-artifact
+      with:
+        name: 'blob-report-${{ github.event.workflow_run.run_attempt }}'
+        path: 'blob-report'
+
+    - name: Merge reports
+      run: |
+        npx playwright merge-reports --reporter html --attachments missing blob-report
+
+    - name: Upload HTML report to Azure
+      run: |
+        REPORT_DIR='run-${{ github.event.workflow_run.id }}-${{ github.event.workflow_run.run_attempt }}'
+        az storage blob upload-batch -s playwright-report -d "\$web/$REPORT_DIR" --connection-string "${{ secrets.AZURE_CONNECTION_STRING }}"
+        echo "Report url: https://mspwblobreport.z1.web.core.windows.net/$REPORT_DIR/index.html"
+
+    - name: Upload blob report to Azure
```

Comment:
Why do we upload the blob report to azure?

Reply:
This was one of the topics for the discussion. Not needed anymore as we are dropping ` --attachments missing` option.